layer/trace_layer.cpp

```diff
@@ -98,6 +98,32 @@ static VkInstance get_instance_handle(const void* handle)
     return (entry != instance_handles.end()) ? entry->second : VK_NULL_HANDLE;
 }
 
+// The vk_layerGetPhysicalDeviceProcAddr of the next layer in the chain.
+// Retrieved during instance creation and forwarded to by this layer's
+// GetPhysicalDeviceProcAddr() after unwrapping its VkInstance parameter.
+static std::mutex                                                    gpdpa_lock;
+static std::unordered_map<VkInstance, PFN_GetPhysicalDeviceProcAddr> next_gpdpa;
+
+static void set_next_gpdpa(const VkInstance instance, PFN_GetPhysicalDeviceProcAddr p_next_gpdpa)
+{
+    assert(instance != VK_NULL_HANDLE);
+    std::lock_guard<std::mutex> lock(gpdpa_lock);
+    next_gpdpa[instance] = p_next_gpdpa;
+}
+
+static PFN_GetPhysicalDeviceProcAddr get_next_gpdpa(const VkInstance instance)
+{
+    assert(instance != VK_NULL_HANDLE);
+    std::lock_guard<std::mutex> lock(gpdpa_lock);
+    auto                        it_gpdpa = next_gpdpa.find(instance);
+    if (it_gpdpa == next_gpdpa.end())
+    {
+        return nullptr;
+    }
+    auto gpdpa = it_gpdpa->second;
+    return gpdpa;
```

Comment:
Very clear, but in the future for stuff this short, I'm okay with just `return it_dpdpa->second`.

Reply:
That's probably an artefact of me debugging this where I had a nullptr assert after this line 123  and could single step and see what I'm dealing with in the locals. Changed.